pkg/resource/snapshot/utility.go

```diff
@@ -12,13 +12,20 @@ import (
 var (
 	condMsgCurrentlyDeleting string = "snapshot currently being deleted"
 	deleteStatus             string = "deleting"
+	copyingStatus            string = "copying"
+	creatingStatus           string = "creating"
+	restoringStatus          string = "restoring"
 )
 
 var (
 	requeueWaitWhileDeleting = ackrequeue.NeededAfter(
 		errors.New("delete is in progress"),
 		ackrequeue.DefaultRequeueAfterDuration,
 	)
+	requeueWaitSnapshotIsReadyForDeleting = ackrequeue.NeededAfter(
+		errors.New("snapshot is not ready for deleting. snapshot must be in either available or failed state in order to be deleted"),
```

Comment:
nit: use string interpolation:
```suggestion
fmt.Errorf("snapshot is not ready for deletion - must be in either %q or %q state", availableStatus, failedStatus),
```

Reply:
Modified.